From 8229937ec299133cf93654b9afc46e5288767ace Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Thu, 1 Mar 2018 21:36:30 +0000
Subject: [PATCH 237/454] overlays: Add updated mmc1 alias to sdio overlays

In the downstream RPi kernel, aliases are used to assign instance
numbers to the SD/MMC interfaces. The updated sdio overlays
effectively rename the device node, so the mmc1 alias has to be
updated in order to preserve the mmc1 numbering, otherwise the device
will appear as mmc2.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/sdio-1bit-overlay.dts | 8 ++++++++
 arch/arm/boot/dts/overlays/sdio-overlay.dts      | 7 +++++++
 2 files changed, 15 insertions(+)

--- a/arch/arm/boot/dts/overlays/sdio-1bit-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sdio-1bit-overlay.dts
@@ -48,6 +48,14 @@
 		};
 	};
 
+	fragment@3 {
+		target-path = "/aliases";
+		__overlay__ {
+			mmc1 = "/soc/sdio@7e300000";
+		};
+	};
+
+
 	__overrides__ {
 		poll_once = <&sdio_1bit>,"non-removable?";
 		sdio_overclock = <&sdio_1bit>,"brcm,overclock-50:0";
--- a/arch/arm/boot/dts/overlays/sdio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sdio-overlay.dts
@@ -48,6 +48,13 @@
 		};
 	};
 
+	fragment@3 {
+		target-path = "/aliases";
+		__overlay__ {
+			mmc1 = "/soc/sdio@7e300000";
+		};
+	};
+
 	__overrides__ {
 		poll_once = <&sdio_ovl>,"non-removable?";
 		bus_width = <&sdio_ovl>,"bus-width:0";
